File: stage-02th/src/main/java/cn/uncleyumo/chapter05/unit06/practise/Test01.java
package cn.uncleyumo.chapter05.unit06.practise;

import java.util.List;

/**
 * @author dev.uncleyumo.cn(BookAir13)
 * @fileName Test01
 * @createDate 4/27/25 April
 * @school Wuxi University
 * @studentID 22344131
 * @description
 * /**
 * 创建一个类，在main方法中完成如下操作
 *   ①创建一个集合，存储多个字符串元素{"张三丰", "张无忌", "张翠山", "王二麻子", "张良", "谢广坤"}
 *   ②使用stream流把集合中所有以"张"开头的元素进行打印
 *   ③使用stream流把"张"开头的集合中的元素长度为3的元素进行遍历打印
*/


public class Test01 {
    public static void main(String[] args) {
        List<String> nameList = List.of("张三丰", "张无忌", "张翠山", "王二麻子", "张良", "谢广坤");
        System.out.println("----------②----------");
        nameList.stream().filter(it -> it.startsWith("张")).forEach(System.out::println);
        System.out.println("----------③----------");
        nameList.stream()
                .filter(it -> it.startsWith("张"))
                .filter(it -> it.length() == 3)
                .forEach(System.out::println);
    }
}

File: stage-02th/src/main/java/cn/uncleyumo/chapter09/unit01/kotlin/Test01.kt
package cn.uncleyumo.chapter09.unit01.kotlin

import cn.uncleyumo.utils.LogPrinter
import java.net.HttpURLConnection
import java.net.InetAddress
import java.net.URL

/**
 * @author dev.uncleyumo.cn(BookAir13)
 * @fileName Test01
 * @createDate 5/12/25 May
 * @school Wuxi University
 * @studentID 22344131
 * @description
 */

fun main() {
//    practise01()
    // 使用示例
    LogPrinter.info("Is Google reachable via HTTP? ${isWebsiteReachable("https://www.google.com")}")
//    LogPrinter.info("Is Baidu reachable via HTTP? ${isWebsiteReachable("https://www.baidu.com")}")
}

fun practise01() {
    val localIP = InetAddress.getLocalHost()
//    LogPrinter.info("Host Name ${localIP.hostName}")
//    LogPrinter.info("Host Address ${localIP.hostAddress}")
//    LogPrinter.info("Canonical Host Name ${localIP.canonicalHostName}")

    val ip2 = InetAddress.getByName("www.baidu.com")
//    LogPrinter.info("Host Name ${ip2.hostName}")
//    LogPrinter.info("Host Address ${ip2.hostAddress}")

    val ip3 = InetAddress.getByName("www.google.com")
    LogPrinter.info("Host Name ${ip3.hostName}")
    LogPrinter.info("Host Address ${ip3.hostAddress}")

//    LogPrinter.info(
//        """
//           localhost can reach ${ip3.hostName}: ${ip3.isReachable(6000)}
//        """.trimIndent()
//    )

    LogPrinter.info(
        """
            localhost can reach ${ip2.hostName}: ${ip2.isReachable(6000)}
        """.trimIndent()
    )
}

fun isWebsiteReachable(urlStr: String, timeout: Int = 10000): Boolean {
    return try {
        val url = URL(urlStr)
        val conn = url.openConnection() as HttpURLConnection
        conn.requestMethod = "HEAD"
        conn.connectTimeout = timeout
        conn.readTimeout = timeout
        val responseCode = conn.responseCode
        println("Response Code: $responseCode")
        responseCode in 200..399
    } catch (e: Exception) {
        LogPrinter.error("Error checking website reachability: ${e.message}")
        false
    }
}
